common/src/main/java/com/microsoft/identity/common/internal/request/AcquireTokenSilentOperationParameters.java

```diff
@@ -54,7 +55,7 @@ public void validate() throws ArgumentException {
 
         if (mAccount == null) {
             Logger.warn(TAG, "The account set on silent operation parameters is NULL.");
-        } else if (!authorityMatchesAccountEnvironment()) {
+        } else if (!isAuthorityB2C() && !authorityMatchesAccountEnvironment()) {
```

Comment:
nit: I'd personally do something like

if (mAccount == null){
 return.
}

if (isAuthorityB2C()){
// Add a comment why we're not verifying.
return
}

if (!authorityMatchesAccountEnvironment()){
throw
}